lib/inputs/base-input.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+class BaseInput {
+    constructor(type, id, params, inputDefaultValue) {
+        this.type = type;
+        this.id = id;
+        this.params = this.normalizeParams(params || {});
```

Comment:
Is there any case when `params` wouldn’t be set? If not, I’d just make it a required parameter.